### ISSUE TITLE
"New Person" open whole form instead of just 'article name'

### ISSUE THREAD

**Issue by natrius** [averi-studios/world-scribe-2-desktop#11]
Currently, looking at category 'Person'

- Press +
- Entering Article Name
- Press Create
- Click on the created person
- edit everything in a bloated interface

Look at like more like the adress book in android.

- Press +
- Enter everything on a new page, press save, done (Maybe even change the button marking. If article is not existing "Create", if article exists already and is just changed "Save")

I may try to look into it if you are interested, but i'm a beginner in JS - you would have to check my changes :D

There are more suggestions, if you like.

**Comment by MarquisLP** [comment]
Thank you for your suggestion!

If I'm understanding correctly, you would like there to be a full "Create Article" form, where the user can enter all details _before_ the Article is actually created? I can picture this, but I'm not sure it would be that different from the current workflow:

1. Enter Article name
2. Confirm creation
3. Enter details on the new page
4. If the user changes their mind, they can just delete the Article

Perhaps I'm not seeing something, but the only benefit I can see with a full form is that the user can cancel before creating the Article. But since Articles can be deleted quickly at any time, I'm not sure it would be that different from creating and then deleting.

***

> Press Create
> Click on the created person

I agree that having to click on the Person afterwards is clunky and a time-waster. In fact, this is a bug that slipped through the cracks: the intended behaviour is to go to the created Person's page right after clicking "Create". However, this only happens when creating an Article from within a Category's page. If, instead, you go the "Articles" page and create a Person from there, it takes you immediately to the created Person's page.

If this bug were to be fixed, so that you're sent to the Person's detail page immediately after clicking "Create", would that be a good enough compromise?

***

Sorry if I misunderstood your suggestion -- please feel free to correct me if anything is off :)

**Comment by natrius** [comment]
Thanks for your quick response :)

It would be better than the current situation, thats for sure, that should be quick fix, i as far as i know. 

To my suggestion: 
If i press + and then could enter the whole data i want to put in right now, i could get redirected to the person overview after the creation. Its faster when creating more people there, in my opinion. '+', enter data with quickly hopping through with `TAB`, 'create', '+', ...

I still think it would be better that way but i fully understand if you just fix the bug and are fine with that. It will be a lot better with that bugfix already :)

**Comment by MarquisLP** [comment]
Oh, I see -- so you would like a "rapid creation" workflow along the lines of:

1. Click "+"
2. Enter Article name and all details
3. Click "Create"
4. Repeat process for all Articles

Is that correct? If so, I can definitely see that being useful! Perhaps instead of having to click "+" to create another Article, the form could have another button named "Confirm and create another", that creates the current Article in the background and resets the form to create another Article.

**Comment by natrius** [comment]
Yes, looks correct. That would help people immensly who would like to transition from something else to worldscribe and have notes somewhere already. 

The additional button sounds like a good idea as well :) For creating multiple it could also a checkbox on the bottom with "Create new person after this" and leave the checkbox activated after pressing save. But i'm not sure if the additional button is still better than that.

And - in my opinion there is no need for an separate "Article name" because it could simply get created from the name. Usally in worlds there is nobody with the same name for clarity.

**Comment by MarquisLP** [comment]
I like the checkbox idea! That way, the user only has to worry about it once, rather than on every Article they create.

If you'd like, I can assign you to this Issue and let you get started with implementing this new feature yourself :)

**Comment by natrius** [comment]
Would love to try, but i'm in the middle of my JS course - it could be this needs a little bit until its finished. Just so you are aware.

**Comment by digulla** [comment]
Instead of the checkbox, I'd opt for a "+" button at the bottom (just like the one that you get in the Articles view) which works like "create another of this category". That would be a natural way to select the category and you could simply add a "Title" field at the top skipping the dialog as @natrius requested.

**Comment by MarquisLP** [comment]
> Would love to try, but i'm in the middle of my JS course - it could be this needs a little bit until its finished. Just so you are aware.

No worries! There's no rush at all, so I'm perfectly okay with keeping this Issue open until you're ready. However, if you'd like me to work on it instead, please let me know :)

**Comment by MarquisLP** [comment]
> Instead of the checkbox, I'd opt for a "+" button at the bottom (just like the one that you get in the Articles view) which works like "create another of this category". That would be a natural way to select the category and you could simply add a "Title" field at the top skipping the dialog as @natrius requested.

Sorry, just to be clear, do you mean adding this new button at the bottom of the "Create Article" dialog? Or at the bottom of the Article page (the one that appears after clicking on an Article's name)?

**Comment by digulla** [comment]
> Sorry, just to be clear, do you mean adding this new button at the bottom of the "Create Article" dialog? Or at the bottom of the Article page (the one that appears after clicking on an Article's name)?

At the bottom of the article itself (i.e. the button on person pages create more persons).

**Comment by MarquisLP** [comment]
> At the bottom of the article itself (i.e. the button on person pages create more persons).

Got it. I agree, this would be a good way to implement rapid Article creation while minimizing the amount of UI work required. What do you think of going with this approach @natrius?
